icu4c/source/common/loclikelysubtags.cpp

```diff
@@ -257,13 +259,15 @@ struct XLikelySubtagsData {
         }
         encoded &= 0x00ffffff;
         encoded %= 27*27*27;
-        char lang[3];
+        char lang[4];
         lang[0] = 'a' + ((encoded % 27) - 1);
         lang[1] = 'a' + (((encoded / 27 ) % 27) - 1);
         if (encoded / (27 * 27) == 0) {
+            lang[2] = '\0';  // Null terminate
             return UnicodeString(lang, 2);
```

Comment:
Issue (pre-existing): this is the slow UnicodeString constructor that pulls in the codepage converter.

https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/classicu_1_1UnicodeString.html#aad4fc33d0aac19c45e73cdf7f844ef5d

You should either build up your UnicodeString directly or use the `US_INV` constructor:

https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/classicu_1_1UnicodeString.html#a613a04db62048c98e32112e5864716db